python/katana/local/_graph.pyx

```diff
@@ -219,45 +219,49 @@ cdef class GraphBase:
 
         return pyarrow_unwrap_table(arrow_table)
 
-    def add_node_property(self, table=None, **kwargs):
+    def add_node_property(self, table=None, TxnContext txn_ctx=None, **kwargs):
         """
         Insert new node properties into this graph.
 
         :param table: A pyarrow Table or other dataframe-like object containing the properties. The table must have
             length ``self.num_nodes()``. (Optional)
+        :param txn_ctx: The tranaction context for passing read write sets.
         :param kwargs: Properties to add. The values must be arrays or sequences of length ``self.num_nodes()``. (Optional)
         """
-        handle_result_void(self.underlying_property_graph().AddNodeProperties(GraphBase._convert_table(table, kwargs)))
+        txn_ctx = TxnContext() if txn_ctx is None else txn_ctx
```

Comment:
I prefer `txn_ctx or TxnContext()`. I admit it's a bit magical, but it really does read clearly. It's either `txn_ctx` or a new context if that's not available, which is very clear once you've seen it. I think it's a good idiom. And it's common in Python.

Reply:
Please apply this throughout the file.